packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -1019,9 +1036,16 @@ export class GarbageCollector implements IGarbageCollector {
 
         // If we are running in GC test mode, delete objects for unused routes. This enables testing scenarios
         // involving access to deleted data.
-        if (this.testMode) {
-            this.runtime.updateUnusedRoutes(gcResult.deletedNodeIds);
+        if (this.sweepDataStoresMode) {
+            const allRoutes = gcResult.deletedNodeIds.concat(gcResult.referencedNodeIds);
+            this.validateSweepRoutes(sweepReadyNodes, allRoutes);
+            const removedRoutes = this.runtime.updateUnusedRoutes(sweepReadyNodes);
```

Comment:
It would be cleaner to have a single function (say this.sweepNodes) that would call updateUnusedRoutes and do whatever updateSweptNodeState does. Also, it could do validateSweepRoutes if an optional argument is passed. This will cosolidate all sweep logic in one place rather than the caller having to call each of these functions.

Reply:
Sure